handler/category-dimension-import.go

```diff
@@ -183,3 +154,45 @@ func (h *CategoryDimensionImport) instanceFailed(ctx context.Context, err error,
 	}
 	return err
 }
+
+// onLastDimension handles the case where all dimensions have been updated to the instance.
+// If the eTag did not change since the last update, then we know that this consumer was the last one to update the instance. In that case:
+// - Set instance to edition-confirmed
+// - Set import job to completed state
+// - send an InstanceComplete kafka message
+func (h *CategoryDimensionImport) onLastDimension(ctx context.Context, e *event.CategoryDimensionImport, eTag string) error {
+
+	// set instance to 'edition-confirmed' state, only if the eTag value did not change
+	_, err := h.datasets.PutInstanceState(ctx, h.cfg.ServiceAuthToken, e.InstanceID, dataset.StateEditionConfirmed, eTag)
+	if err != nil {
+		switch putErr := err.(type) {
+		case *dataset.ErrInvalidDatasetAPIResponse:
+			if putErr.Code() == http.StatusConflict {
+				log.Info(ctx, "instance state could not be set to 'edition-confirmed' because it changed between putting the last dimension option and counting")
```

Comment:
This log message and the comment on the next line don't seem to be saying the same thing.
Does the log message also need to say that 'things' are actually OK (assuming i'm understanding what is going on here and things are indeed actually OK)

Reply:
Yes, good catch! The log doesn't say that this is a valid scenario. I will modify the message.

Reply:
Hmmm ...
If the instance state could not be set to 'edition-confirmed' ...
Should it check that the state has been set to 'edition-confirmed' by some other instance of API, and if its not in the desired state, should it at least retry to set it a few more times ?

Reply:
so, for example:
- service instance A fails to set the state to edition confirmed (or fails to set import job to complete)
- service instance B checks that all dimension options have been processed, but the state did not change, so it changes it.
I think it would work, my main worry was that we could potentially produce 2 or more kafka messages, but if instance A aborted due to some issue before then, it did not send the kafka message, so it woudld be safe for instance B to try to change states and send the kafka message.
Thanks! I will do that.

Reply:
About the API calls retries, I think the client in dp-net (used by dp-api-clients-go) already implements some retries